packages/client/src/internals/httpRequest.ts

```diff
@@ -59,17 +59,23 @@ export function httpRequest<TResponseShape = TRPCResponse>(
     const url = getUrl();
 
     Promise.resolve(rt.headers())
-      .then((headers) =>
-        rt.fetch(url, {
+      .then((headers) => {
+        Object.keys(headers).forEach((key) => {
+          if (headers[key] === undefined) {
+            delete headers[key];
```

Comment:
I like this solution but one shouldn't mutate an argument from a function

Reply:
Instead, create a new object

Reply:
Good point! Have amended 🎉